ivy/utils/exceptions.py

```diff
@@ -390,6 +396,45 @@ def _handle_exceptions(*args, **kwargs):
             raise ivy.utils.exceptions.IvyBackendException(
                 fn.__name__, str(e), include_backend=True
             )
+        except InplaceUpdateException as e:
+            _configure_stack_trace(e.__traceback__)
+            raise ivy.utils.exceptions.InplaceUpdateException(
+                fn.__name__, str(e), include_backend=True
+            )
 
     _handle_exceptions.handle_exceptions = True
     return _handle_exceptions
+
+
+# Inplace Update
+
+
+def _handle_inplace_mode():
+    current_backend = ivy.current_backend_str()
+    if not ivy.native_inplace_support and ivy.inplace_mode == "lenient":
+        warnings.warn(
+            f"The current backend: '{current_backend}' does not support "
+            "inplace updates natively. Ivy would quietly create new arrays when "
+            "using inplace updates with this backend, leading to memory overhead "
+            "(same applies for views). If you want to control your memory "
+            "management, consider doing ivy.set_inplace_mode('strict') which "
+            "should raise an error whenever an inplace update is attempted "
+            "with this backend."
+        )
+
+
+def _check_inplace_update_support(ensure_in_backend, x):
```

Comment:
I think having `x` as the first argument and `ensure_in_backend` as the second argument would be more intuitive